docs/user-guide/spark-tool.rst

```diff
@@ -0,0 +1,184 @@
+GraphAr Spark Tools
+============================
+
+Overview
+-----------
+
+GraphAr Spark tools are provided as a library for generating, loading and transforming GAR files with Apache Spark easy. It consists of the following parts:
+
+- **Information Classes**: As same with in C++ SDK, the information classes are implemented as a part of Spark tools for constructing and accessing the meta information about the graphs, vertices and edges in GraphAr.
+- **IndexGenerator**: The IndexGenerator helps to generate the indices for vertex/edge DataFrames. In most cases, IndexGenerator is first utilized to generate the indices for a DataFrame (e.g., from primary keys), and then this DataFrame can be written into GAR files through the Writer.
+- **Writer**: The GraphAr Spark Writer provides a set of interfaces that can be used to write Spark DataFrames into GAR files. Every time it takes a DataFrame as the logical table for a type of vertices or edges, assembles the data in specified format (e.g., reorganize the edges in the CSR way) and then dumps it to standard GAR files (orc, parquet or CSV files) under the specific directory path.
+- **Reader**: The GraphAr Spark Reader provides a set of interfaces that can be used to read GAR files. It reads a set of vertices or edges at a time and assembles the result into Spark DataFrames. Similar with the Reader SDK in C++, it supports the users to specify the data they need, e.g., to read a single property group instead of all properties.
+ 
+Use Cases
+----------
+
+The GraphAr Spark Tools can be applied to these scenarios:
+
+- Transform data between GAR and other data sources (e.g., Hive, Neo4j, NebulaGraph, ...).
+- Transform GAR data between different file types (e.g., from CSV to parquet).
+- Transform GAR data between different adjList types (e.g., from COO to CSR).
+- Modify existing GAR data (e.g., add new vertices/edges).
+
+
+Get GraphAr Spark Tools
+------------------------------
+
+Make the graphar-spark-tools directory as the current working directory:
+
+.. code-block:: shell
+
+   cd spark/
+
+Compile package:
+
+.. code-block:: shell
+
+   mvn package
+
+After compilation, a similar file *graphar-0.1.0-SNAPSHOT.jar* is generated in the directory *spark/target/*.
```

Comment:
Nits: maybe encourage user to use `graphar-0.1.0-SNAPSHOT-shaded.jar` is better? the shaded jar include all dependencies of graphar spark.